app/models/subjects/ucas_subject_to_dfe_subject_mapping_collection.rb

```diff
@@ -0,0 +1,21 @@
+module Subjects
+  class UCASSubjectToDFESubjectMappingCollection
+    def initialize(config:)
```

Comment:
It feels like the majority of the work is done via the config. Should the config live in here and we just pass in the level?

Reply:
Also, maybe make this take a collection and then iterate over it, using the other class to get the result one by one rather than using a collection again?

Reply:
My reasoning for splitting the data and the functionality was:

- to keep this class easier to unit test and reason about
- have all the specific "settings" defined in the config

kinda like the model we have with the `settings.yml` files.

I can combine them if you think it's absolutely needed.